### PR TITLE
add update item

### PR DESCRIPTION
**Pull request by fred0405**
File change:
	service/models.py
	ervice/routes.py
	tests/test_routes.py

Add update item and testcases

### CODE REVIEW

File: service/models.py [hunk 1]
```diff
@@ -304,8 +304,18 @@ def find_by_wishlist_id(cls, wishlist_id):
...
         logger.info("Processing owner id query for %s ...", str(wishlist_id))
         return cls.query.filter(cls.wishlist_id == wishlist_id)
-
-   
+    
+    @classmethod
+    def find_by_wishlist_and_item_id(cls, wishlist_id, item_id):
+        """Returns all Item with the given owner id
+        Args:
+            name (string): the name of the WishlistModels you want to match
```
Comment (sakshamgarg) [inline service/models.py:312]: kindly update the argument list according to the function written.

File: service/routes.py [hunk 2]
```diff
@@ -112,6 +112,38 @@ def update_wishlist(wishlist_id):
...
+    Update an item from a wishlist
+    """
+    app.logger.info("Request to update an Item %s from Wishlist with id: %s", item_id, wishlist_id)
+
+    # See if the wishlist exists and abort if it doesn't
+    wishlist = Wishlist.find(wishlist_id)
+    if not wishlist:
+        abort(
+            status.HTTP_404_NOT_FOUND,
+            f"Order with id '{wishlist_id}' was not found.",
```
Comment (sakshamgarg) [inline service/routes.py:128]: The error message has wrong message

File: tests/test_routes.py [hunk 3]
```diff
@@ -309,3 +309,86 @@ def test_delete_item_nonexistent_wishlist(self):
...
+
+        # add item
+        resp = self.app.post(
+            f"{BASE_URL}/{wishlist.id}/items",
+            json=item.serialize(),
+            content_type="application/json",
+        )
+        self.assertEqual(resp.status_code, status.HTTP_201_CREATED)
+
+        # update item with wrong order id
```
Comment (sakshamgarg) [inline tests/test_routes.py:383]: The comment has wrong message

File: tests/test_routes.py [hunk 4]
```diff
@@ -309,3 +309,86 @@ def test_delete_item_nonexistent_wishlist(self):
...
+        updated_item.wishlist_id = item.wishlist_id + 123
+        updated_item.id = item.id
+        resp = self.app.put(
+            f"{BASE_URL}/{updated_item.wishlist_id}/items/{updated_item.id}",
+            json=updated_item.serialize(),
+            content_type="application/json",
+        )
+        self.assertEqual(resp.status_code, status.HTTP_404_NOT_FOUND)
+        data = resp.get_json()
+        self.assertEqual(data["message"], f"404 Not Found: Order with id '{updated_item.wishlist_id}' was not found.")
```
Comment (sakshamgarg) [inline tests/test_routes.py:394]: The error message has wrong message

**Review by sakshamgarg — CHANGES_REQUESTED** [review]
Kindly add the test report to the PR.

**Review by sakshamgarg — APPROVED** [review]
LGTM

**Review by reubenabraham — APPROVED** [review]
LGTM. Merging.

**Pull request merged**